### ISSUE TITLE
Docker image for front-end should not be broken

### ISSUE THREAD

**Issue by frol** [near/near-explorer#843]
## Problem

@luixo After #824, our Docker image for the front-end side does not work due to (1) non-existing next.config.ts file in the final image (2) no TS runtime on the final stage of the image.

I believe we should transpile `next.config.ts` on the build stage into `next.config.js`, so we don't introduce typescript runtime.

## How did we discover this problem?

We had our deployment of the front-end side broken for a few days.

**Comment by luixo** [comment]
Actually, moving `ts-node` to `dependencies` doesn't introduce any typescript runtime (where runtime is either running server or client code) and it seems the simplest solution to me (but we'll have to keep aware that we have `ts-node` in our build deps).

Otherwise we'll have to be pretty tricky with `next.config.ts` / `next.config.js`, like making condition based on actual [phase](https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/constants.ts#L1-L5) or rewriting `next.config.js` every time we're up for a production build (doing this multiple times locally is not convenient)

**Comment by frol** [comment]
Can we just transpile next.config.ts into next.config.js as one of the Dockerfile stage?

**Comment by frol** [comment]
> Is there a downside of getting ts-node to dependencies?

That just looks odd to me.

> We can, but we'll have to write a separate command for that

We do many separate commands there to ensure that we don't bring dev-dependencies into production, so I don't see it as a problem.

> and next.config.js will eventually be overriden.

Which is good, I believe

**Comment by luixo** [comment]
> We do many separate commands there to ensure that we don't bring dev-dependencies into production, so I don't see it as a problem.

That's not much of a problem to have one more command, I just feel it smelly that it modifies git-registered files.
I don't get what command(s) ensure that we don't bring dev-dependencies into production (only "build" probably?).

> Which is good, I believe

I believe that an overriden file (that exist in git repo in a different version) is inherently bad.

I'll implement an option with an overriden file. Let's continue this discussion in PR.

**Comment by frol** [comment]
> I just feel it smelly that it modifies git-registered files.

We create a production build here. If this is the concert, we may name it differently or put the transpiled file into a different folder. We don't keep git when create a second stage of the image, so I would treat the second stage as an artifact rather than a project folder with git.

> I don't get what command(s) ensure that we don't bring dev-dependencies into production (only "build" probably?).

All of these:

```
RUN npm install --prod
RUN npm run build

...

COPY --from=build /near-explorer/production_node_modules ./node_modules
COPY --from=build /near-explorer/frontend/public ./public
COPY --from=build /near-explorer/frontend/next.config.js ./
COPY --from=build /near-explorer/frontend/.next ./.next
COPY --from=build /near-explorer/frontend/package.json ./package.json
```

**Comment by frol** [comment]
Resolved in #856
